src/hooks/useEditorView.ts

```diff
@@ -1,56 +1,33 @@
-import type { EditorState, Transaction } from "prosemirror-state";
+import { Schema } from "prosemirror-model";
+import { EditorState } from "prosemirror-state";
+import type { Plugin, Transaction } from "prosemirror-state";
 import { EditorView } from "prosemirror-view";
-import type { DirectEditorProps } from "prosemirror-view";
-import { useLayoutEffect, useState } from "react";
+import type { EditorProps } from "prosemirror-view";
+import { useLayoutEffect, useMemo, useState } from "react";
 import { flushSync } from "react-dom";
 
-import { useForceUpdate } from "./useForceUpdate.js";
+import type { EditorContextValue } from "../contexts/EditorContext.js";
 
-function withFlushedUpdates<This, T extends unknown[]>(
-  fn: (this: This, ...args: T) => void
-): (...args: T) => void {
-  return function (this: This, ...args: T) {
-    flushSync(() => {
-      fn.call(this, ...args);
-    });
-  };
-}
+import { useComponentEventListeners } from "./useComponentEventListeners.js";
 
-function defaultDispatchTransaction(this: EditorView, tr: Transaction) {
-  this.updateState(this.state.apply(tr));
-}
+const EMPTY_SCHEMA = new Schema({
+  nodes: {
+    doc: { content: "text*" },
+    text: { inline: true },
+  },
+});
 
-type EditorStateProps =
-  | {
-      state: EditorState;
-    }
-  | {
-      defaultState: EditorState;
-    };
+const EMPTY_STATE = EditorState.create({
+  schema: EMPTY_SCHEMA,
+});
 
-export type EditorProps = Omit<DirectEditorProps, "state"> & EditorStateProps;
+let didWarnValueDefaultValue = false;
 
-function withFlushedDispatch(
-  props: EditorProps,
-  forceUpdate: () => void
-): EditorProps & {
-  dispatchTransaction: EditorView["dispatch"];
-} {
-  return {
-    ...props,
-    ...{
-      dispatchTransaction: function dispatchTransaction(
-        this: EditorView,
-        tr: Transaction
-      ) {
-        const flushedDispatch = withFlushedUpdates(
-          props.dispatchTransaction ?? defaultDispatchTransaction
-        );
-        flushedDispatch.call(this, tr);
-        if (!("state" in props)) forceUpdate();
-      },
-    },
-  };
```

Comment:
It's easier to write this to update state through React by inlining it in the component. It provides a nice narrative structure wherein the component handles each of the direct editor props in turn (`state`, `plugins`, and `dispatchTransaction`). It's easier to see that the props overall aren't straightforward to memoize without some shallow comparisons, and we are better off delegating that to ProseMirror itself, I think.